core/src/lib/runscript.cc

```diff
@@ -134,41 +134,49 @@ int RunScripts(JobControlRecord* jcr,
           NSTDPRNT(script->command));
     runit = false;
 
-    if ((script->when & SCRIPT_Before) && (when & SCRIPT_Before)) {
-      if ((script->on_success &&
-           (jcr->JobStatus == JS_Running || jcr->JobStatus == JS_Created)) ||
-          (script->on_failure &&
-           (JobCanceled(jcr) || jcr->JobStatus == JS_Differences))) {
-        Dmsg4(200, "runscript: Run it because SCRIPT_Before (%s,%i,%i,%c)\n",
-              script->command.c_str(), script->on_success, script->on_failure,
-              jcr->JobStatus);
-        runit = true;
+    if (!script->IsLocal()) {
```

Comment:
As far as I can see, currently, every runscript is by default not local (runscript->target is not empty). As a matter of fact this statement is always true and the following else won't be reached.

Reply:
No, as discussed in PR #358.